### PR TITLE
rusty patcher

### PR DESCRIPTION
**Pull request by dzshn**
:transgender_flag:

### CODE REVIEW

**Comment by Vendicated** [comment]
unrust immediately

File: qpatcher/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+                explode!(format!("{:?}", e).as_str(), "run as root!")
+            }
+            Err(e) => explode!(format!("{:?}", e).as_str()),
+        }
+    }
+    if patch {
+        if download {
+            // Iterate instead of using create_dir_all so we can fix perms
+            // ::<> fishyy
+            for p in quartet_path.ancestors().collect::<Vec<_>>().iter().rev() {
```
Comment (fawni) [inline qpatcher/src/main.rs:240]:
```suggestion
            for p in quartet_path.ancestors().into_iter().rev() {
```
NOW

**Review by fawni — CHANGES_REQUESTED** [review]

File: qpatcher/src/main.rs (hunk 1 above)
Comment (Vap0r1ze) [reply qpatcher/src/main.rs:240]: byee fishyyyy

Comment (fawni) [reply qpatcher/src/main.rs:240]: fishy stays

**Review by Vap0r1ze — APPROVED** [review]
im so tired but i think its epic maybe group the functions by target in `path.rs` tho

**Pull request merged**